update iOS structure

Maxwell-Yang-PDFTron opened the pull request.

JamesTron reviewed (commented):
I haven't looked at this in detail, because I believe it's just changing some setup stuff and then copy/paste, is that right? If so, I will approve. If there is something you are unsure of please tell me.

Maxwell-Yang-PDFTron commented:
> I haven't looked at this in detail, because I believe it's just changing some setup stuff and then copy/paste, is that right? If so, I will approve. If there is something you are unsure of please tell me.

I have renamed the two files containing FluttterDocumentView into DocumentViewFactory since the documentView is no longer required. Else than that, I also deleted PTPluginUtils and moved all of the constants/functions back into PTFlutterPlugin. I believe that is all and nothing should break.

JamesTron commented on ios/Classes/DocumentViewFactory.m:
```diff
@@ -0,0 +1,28 @@
+    return [FlutterStandardMessageCodec sharedInstance];
+}
+
+- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args
```
args is completely ignored here?

Maxwell-Yang-PDFTron replied:
It was not used before in DocumentView as well, so yeah

JamesTron commented on ios/Classes/DocumentViewFactory.m:
```diff
@@ -0,0 +1,28 @@
+
+- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args
+{
+    return [PdftronFlutterPlugin registerWithFrame:frame viewIdentifier:viewId messenger:_messenger];
```
You should use `self.messenger`, not `_messenger`. Generally only use the backing variable (`_messenger`) directly in the initializer (where it is done this way because otherwise you can get into trouble when using inheritance.)

JamesTron commented on ios/Classes/DocumentViewFactory.m:
```diff
@@ -0,0 +1,28 @@
+#import "PdftronFlutterPlugin.h"
+#import "PTPluginUtils.h"
+
+@implementation DocumentViewFactory {
```
This seems like an odd class to me. Why does it exist? One method calls a static method on another class. The other method calls a static method too, but uses this class' one member (instance) variable. I guess the callers don't have access to the `FlutterBinaryMessenger` when they call the create method?

Maxwell-Yang-PDFTron replied:
I deleted that class and merged it into PTFlutterPlugin

JamesTron commented on ios/Classes/PdftronFlutterPlugin.m:
```diff
@@ -574,6 +572,23 @@ - (FlutterError* _Nullable)onCancelWithArguments:(id _Nullable)arguments
+}
+
+- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args
+{
```
```NSAssert(self.messenger, @"Messenger cannot be nil");```

JamesTron commented on ios/Classes/DocumentViewFactory.m:
```diff
@@ -2,15 +2,13 @@
     self = [super init];
     if (self) {
-        _messenger = messenger;
+        self.messenger = messenger;
```
Should use `_messenger` in the initializer.
